### ISSUE TITLE
Error: multiz.v11.2: ckalloc: request for negative space.

### ISSUE THREAD

**Issue by Zhuxitong** [multiz/multiz#4]
Could you please tell me what would cause the error below?

> multiz.v11.2: ckalloc: request for negative space.

Some of my MAF files ran good, but others had this error. I could not fingure out what difference between them..

**Comment by rsharris** [comment]
@Zhuxitong This appears to be the same as https://github.com/multiz/multiz/issues/2 .

**Comment by Zhuxitong** [comment]
Thanks for your quick reply. 

The problem doesn't seem to be solved there. An alternative solution  was made rather than mergeing multiple MAF files. That might be good. I will try that. And I will also check why some of my MAF files were successfully merged but others not.

**Comment by rsharris** [comment]
Right, the problem wasn't solved. The original author hasn't working on multiz in over a decade, and I don't have the bandwidth to fix problems in this package.

The reason it happens for some of your runs and not for others is most likely that those that fail have more alignable content than the others do. Thus they try to allocate more memory, and the variable that determines how much memory is needed overflows. That's a guess — it's not clear where in the program the failing use of ckalloc() is called from. This could be determined but it would take time and effort I can't devote to this.

If you're ambitious, you could try the potential "fix" I mentioned in issue 2. No guarantee that will work though.

**Comment by Zhuxitong** [comment]
@rsharris yes, you are right about the problem that some MAFs failed. I used maf-cut to select only a short of alignable region, like 2000bp, and multiz worked then. 

Some of my species are very close to the reference thus very long alignable regions were discovered. They are the problematic ones. So, I will probably first cut the interseting region out and then merge them for a better visualization.

Anyway, thanks for your kind reply and I think this might be useful for others as more and more close related complete genomes are finished now, it's a time to think how to compare them in a different way.

**Comment by rsharris** [comment]
@Zhuxitong I'm not positive, but I believe the UCSC browser team uses a modified version of multiz. If so, they may have resolved problems like this one. You might also consider their cactus aligner.

For closely-related assemblies (e.g. individuals of the same species) the metagenomics community probably has newer comparison tools. That community may even have tools for closely-related species comparison as well.

Regarding newer assemblies of closely-related species, the biggest problem the multiz pipeline will have is that it depends on pairwise alignments from lastz (which I wrote). The newer assemblies contain telemeric and centromeric repeats that earlier assemblies avoided. And such repeats can cause lastz to grind to a halt.
